zarr/util.py

```diff
@@ -10,13 +11,121 @@
 from asciitree import BoxStyle, LeftAligned
 from asciitree.traversal import Traversal
 from collections.abc import Iterable
-from numcodecs.compat import ensure_ndarray, ensure_text
+from numcodecs.compat import ensure_text
 from numcodecs.registry import codec_registry
 from numcodecs.blosc import cbuffer_sizes, cbuffer_metainfo
 
 from typing import Any, Callable, Dict, Optional, Tuple, Union
 
 
+# TODO: move the function to numcodecs
+def ensure_ndarray(buf):
```

Comment:
Were there other changes here outside the `try...except...` for CuPy below?

Reply:
No, this is the only change